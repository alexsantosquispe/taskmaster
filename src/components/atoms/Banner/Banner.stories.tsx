import type { Meta, StoryObj } from '@storybook/react';

import { Banner } from './Banner';

const meta = {
  title: 'Atoms/Banner',
  component: Banner
} satisfies Meta<typeof Banner>;

export default meta;

type Story = StoryObj<typeof Banner>;

export const Default: Story = () => {
  return <Banner message="test" description="Lorem ipsum dolor sit amet" />;
};

Default.args = {};
