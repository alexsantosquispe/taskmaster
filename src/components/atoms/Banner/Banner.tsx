interface BannerProps {
  message: string;
  description?: string;
}

export const Banner = ({ message, description }: BannerProps) => {
  return (
    <section className="dark:bg-primary flex min-h-80 flex-col items-center justify-center gap-1 rounded-xl border border-neutral-200 bg-neutral-50 p-8 dark:border-white/15">
      <h3 className="text-2xl font-bold">{message}</h3>
      {description && (
        <p className="text-xl font-semibold text-neutral-500 dark:text-neutral-400">
          {description}
        </p>
      )}
    </section>
  );
};
