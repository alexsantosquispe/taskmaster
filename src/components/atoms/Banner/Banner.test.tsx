import '@testing-library/jest-dom';

import { render, screen } from '@testing-library/react';

import { Banner } from './Banner';

describe('Banner', () => {
  it('should render the component correctly', () => {
    const component = render(
      <Banner message="test" description="description" />
    );

    expect(component).toMatchSnapshot();

    const title = screen.getByText('test');
    const description = screen.getByText('description');

    expect(title).toBeInTheDocument();
    expect(description).toBeInTheDocument();
  });
});
