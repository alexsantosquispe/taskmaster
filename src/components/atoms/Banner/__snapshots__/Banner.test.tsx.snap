// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Banner should render the component correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="dark:bg-primary flex min-h-80 flex-col items-center justify-center gap-1 rounded-xl border border-neutral-200 bg-neutral-50 p-8 dark:border-white/15"
      >
        <h3
          class="text-2xl font-bold"
        >
          test
        </h3>
        <p
          class="text-xl font-semibold text-neutral-500 dark:text-neutral-400"
        >
          description
        </p>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="dark:bg-primary flex min-h-80 flex-col items-center justify-center gap-1 rounded-xl border border-neutral-200 bg-neutral-50 p-8 dark:border-white/15"
    >
      <h3
        class="text-2xl font-bold"
      >
        test
      </h3>
      <p
        class="text-xl font-semibold text-neutral-500 dark:text-neutral-400"
      >
        description
      </p>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
